qa/L0_server_status/server_status_test.py

```diff
@@ -131,57 +151,94 @@ def test_model_latest_infer(self):
-                    self.assertEqual(len(server_status1.model_status[model_name].version_status), 3,
-                                    "expected status for 3 versions for model " + model_name)
+                for pair in [("localhost:8000", "http"), ("localhost:8001", "grpc")]:
+                    if pair[1] == "http":
+                        triton_client = httpclient.InferenceServerClient(
+                            url=pair[0], verbose=True)
+                    else:
+                        triton_client = grpcclient.InferenceServerClient(
+                            url=pair[0], verbose=True)
+
+                    self.assertTrue(triton_client.is_server_live())
+                    self.assertTrue(triton_client.is_server_ready())
+                    model_metadata = triton_client.get_model_metadata(
+                        model_name)
+
+                    if pair[1] == "http":
+                        self.assertEqual(model_name, model_metadata['name'])
+                    else:
+                        self.assertEqual(model_name, model_metadata.name)
+
                     for v in (1, 2, 3):
-                        self.assertTrue(v in server_status1.model_status[model_name].version_status,
-                                        "expected version " + str(v) + " status for model " + model_name)
-                        self.assertEqual(server_status1.model_status[model_name].version_status[v].ready_state,
-                                        server_status.MODEL_READY)
+                        self.assertTrue(triton_client.is_model_ready(
+                            model_name, model_version=str(v)))
 
                     # Only version 3 should have infer stats
                     for v in (1, 2, 3):
-                        version_status = server_status1.model_status[model_name].version_status[v]
+                        infer_stats = triton_client.get_inference_statistics(model_name, model_version=str(v))
                         if v == 3:
-                            self.assertEqual(len(version_status.infer_stats), 1,
-                                            "expected 1 infer stats for v" + str(v) + " model " + model_name)
-                            self.assertTrue(1 in version_status.infer_stats,
-                                            "expected batch 1 status for v" + str(v) + " model " + model_name)
-                            infer_stats = version_status.infer_stats[1]
-                            self.assertTrue(infer_stats.success.count, 1)
+                            if pair[1] == "http":
+                                self.assertEqual(len(infer_stats['model_stats']), 1,
+                                                "expected 1 infer stats for version " + str(v) +
+                                                 " of model " + model_name)
+                                stats = infer_stats['model_stats'][0]['inference_stats']
```

Comment:
[0]  You can't rely on the model stats being in order by version. You should instead verify that you get back 3 sets of stats and then loop over those stats. If the stat is for version 3 then check for non-zero success count, if version 1 or 2 check for zero success count.